### PR TITLE
make provider_name forced new to avoid patch problems

### LINKED ISSUE #42: Upgrade from M2 to M10 fails
Upgrade from M2 to M10 fails

It does not seem to be possible to upgrade an M2 cluster to an M10 cluster even though it's possible via the Atlas UI (and I assume via the API).

I have attached two tf files that show the cluster definitions before and after. Both can be used to create new clusters without any error.

To reproduce, use the following commands:

    export MONGODB_ATLAS_PUBLIC_KEY="abcd"
    export MONGODB_ATLAS_PRIVATE_KEY="abcd-abcd"
    export TF_VAR_project_id="1234"
    
    cp main.tf.1 main.tf 
    terraform init
    terraform apply
    
    cp main.tf.2 main.tf
    terraform apply
    
This fails with the error message "400 (request "Bad Request") Invalid attribute backingProviderName specified." which leads me to speculate that even though backing_provider_name is not set in the main.tf.2 it somehow gets transferred to the atlas API.

If I do instead

     terraform taint mongodbatlas_cluster.issue  
     terraform apply
     
it works but (of course) the data in the cluster is lost.

[main.tf.zip](https://github.com/terraform-providers/terraform-provider-mongodbatlas/files/3676905/main.tf.zip)

### PR DESCRIPTION
**Pull request by marinsalinas**
Closes #42 

fixed the issue submitted by @therealppa.

### CODE REVIEW

File: mongodbatlas/resource_mongodbatlas_cluster_test.go [hunk 1]
```diff
@@ -251,6 +251,49 @@ func TestAccResourceMongoDBAtlasCluster_importBasic(t *testing.T) {
...
+				Config: testAccMongoDBAtlasClusterConfigTenant(projectID, name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckMongoDBAtlasClusterExists(resourceName, &cluster),
+					testAccCheckMongoDBAtlasClusterAttributes(&cluster, name),
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					// resource.TestCheckResourceAttr(resourceName, "disk_size_gb", "100"),
+					// resource.TestCheckResourceAttrSet(resourceName, "mongo_uri"),
+					// resource.TestCheckResourceAttrSet(resourceName, "replication_specs.#"),
+					// resource.TestCheckResourceAttrSet(resourceName, "replication_specs.0.regions_config.#"),
```
Comment (PacoDw) [inline mongodbatlas/resource_mongodbatlas_cluster_test.go]: Could you remove the commented code?

File: mongodbatlas/resource_mongodbatlas_cluster_test.go [hunk 2]
```diff
@@ -251,6 +251,49 @@ func TestAccResourceMongoDBAtlasCluster_importBasic(t *testing.T) {
...
+				Config: testAccMongoDBAtlasClusterConfigTenantUpdated(projectID, name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckMongoDBAtlasClusterExists(resourceName, &cluster),
+					testAccCheckMongoDBAtlasClusterAttributes(&cluster, name),
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					// resource.TestCheckResourceAttr(resourceName, "disk_size_gb", "100"),
+					// resource.TestCheckResourceAttrSet(resourceName, "mongo_uri"),
+					// resource.TestCheckResourceAttrSet(resourceName, "replication_specs.#"),
+					// resource.TestCheckResourceAttrSet(resourceName, "replication_specs.0.regions_config.#"),
```
Comment (PacoDw) [inline mongodbatlas/resource_mongodbatlas_cluster_test.go]: The same that above comment

**Review by PacoDw — APPROVED** [review]
Very Nice! Thanks @marinsalinas

**Review by themantissa — APPROVED** [review]
LGTM - nicely done!

**Pull request merged**